whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/FullTextSearchService.java

```diff
@@ -65,6 +68,36 @@ public Response search(
         }
     }
 
+    private String escapeColon(final String query){
+        if (query==null || !query.contains(":")){
+            return query;
+        }
+
+        final StringBuilder sb = new StringBuilder();
+
+        final Pattern colonRegex = Pattern.compile("[^:\\n]*((:){1,2}|$)"); //take everything that finish with :
+        final Matcher wordsWithTrailingColon = colonRegex.matcher(query);
```

Comment:
Just split by colon ? Would be much more readable and easier ?

Reply:
That was my first attempt. If you split by colon you lose a lot of information. You can have as well ::/64 for example, if you split that you will have ["", "",] you don't know what to do.

Another example 2a00:1f78::fffe/48 -> [2a00, 1f78, "", fffe]
with this information you can do anything.

So the idea here is to get everything until the colon (1 or 2 and with them included), so you can figure out if that colon is already escaped or not and then you can.

On other hand, we are building another query, not changed the one that we already have...if you split by colon you will lose the original piece of the query.

Reply:
Talked separately to simplify this using split